src/discovery/HazelcastCloudAddressProvider.ts

```diff
@@ -37,4 +39,31 @@ export class HazelcastCloudAddressProvider implements AddressProvider {
             return [];
         });
     }
+
+    translate(address: Address): Promise<Address> {
+        if (address == null) {
+            return Promise.resolve(null);
+        }
+        const publicAddress = this.privateToPublic.get(address.toString());
+        if (publicAddress != null) {
+            return Promise.resolve(publicAddress);
+        }
+
+        return this.refresh().then(() => {
+            if (this.privateToPublic.get(address.toString())) {
```

Comment:
nit: `this.privateToPublic.get(address.toString())` is read twice here, which can be avoided if `publicAddress` would be made a `let` variable and read in here.